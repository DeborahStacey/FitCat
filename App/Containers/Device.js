import React from 'react'
import { AsyncStorage, Text, View, ScrollView } from 'react-native'
import { default as StorageKeys } from '../Config/StorageKeys'
import Icon from 'react-native-vector-icons/FontAwesome'
import Moment from 'moment'

import styles from './Styles/DeviceStyle'

export default class Device extends React.Component {
  constructor (props) {
    super(props)
    this.state = {
      device: [],
      devicePresent: false
    }
    this.fetchDeviceData()
  }

  async fetchDeviceData () {
    let accessToken = await AsyncStorage.getItem(StorageKeys.FITBIT_ACCESS_TOKEN)

    fetch(`https://api.fitbit.com/1/user/-/devices.json`, {
      headers: {
        'Accept': 'application/json',
        'Content-Type': 'application/json',
        'Authorization': 'Bearer ' + accessToken
      }
    }).then((response) => response.json()).then((responseJson) => {
      if (responseJson.length !== 0) {
        this.setState({
          device: responseJson[0], // If we have more than one fitbit this will need to change..
          devicePresent: true
        })
      }
    })
  }

  matchesBatteryLevel = (toMatch) => {
    return this.state.device.battery === toMatch
  }

  render () {
    return (
      <View style={styles.mainContainer}>
        <ScrollView style={styles.container}>
          { this.matchesBatteryLevel('High') ? <Icon name='battery-full' style={styles.fullBattery} /> : null }
          { this.matchesBatteryLevel('Medium') ? <Icon name='battery-three-quarters' style={styles.mediumBattery} /> : null }
          { this.matchesBatteryLevel('Low') ? <Icon name='battery-quarter' style={styles.lowBattery} /> : null }
          { this.matchesBatteryLevel('Empty') ? <Icon name='battery-empty' style={styles.emptyBattery} /> : null }
          <Text style={styles.sectionText}>
            Device type: { this.state.device.deviceVersion }
          </Text>
          <Text style={styles.sectionText}>
            Battery level: { this.state.device.battery }
          </Text>
          <Text style={styles.sectionText}>
            Last synced: { Moment(this.state.device.lastSyncTime).format('MMMM Do YYYY, h:mm:ss a') }
          </Text>
        </ScrollView>
      </View>
      )
  }
}
