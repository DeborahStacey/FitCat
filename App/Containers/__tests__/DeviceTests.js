import React from 'react'
import Device from '../Device'
import Moment from 'Moment'

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer'

jest.mock('../../Services/MomentWrapper', () => {
  return {
    getMomentFromDate: () => { return new Moment('20161212') }
  }
})
it('renders the device page correctly', () => {
  const tree = renderer.create(
    <Device />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})
