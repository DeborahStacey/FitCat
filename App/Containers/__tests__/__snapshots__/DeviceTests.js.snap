exports[`test renders the device page correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#F1F5FF",
      "flex": 1,
      "marginTop": 64,
    }
  }>
  <ScrollView
    style={
      Object {
        "flex": 1,
        "paddingTop": 10,
      }
    }>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#2C457F",
          "fontWeight": "bold",
          "marginVertical": 5,
          "textAlign": "center",
        }
      }>
      Device type: 
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#2C457F",
          "fontWeight": "bold",
          "marginVertical": 5,
          "textAlign": "center",
        }
      }>
      Battery level: 
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#2C457F",
          "fontWeight": "bold",
          "marginVertical": 5,
          "textAlign": "center",
        }
      }>
      Last synced: 
      December 12th 2016, 12:00:00 am
    </Text>
  </ScrollView>
</View>
`;
